File: src/main/java/de/aquafun3d/challenge/listeners/DmgListener.java
package de.aquafun3d.challenge.listeners;

import de.aquafun3d.challenge.Main;
import de.aquafun3d.challenge.utils.Utils;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.entity.EntityDamageByEntityEvent;
import org.bukkit.event.entity.EntityDamageEvent;

import java.util.Objects;

public class DmgListener implements Listener {

	@EventHandler
	public void onDmg(EntityDamageEvent e) {
		if (e.getEntity() instanceof Player) {
			e.setDamage(e.getDamage() * Main.settingsInv.getDamage());
			if (Utils.dmgListenerToggle) {
				String cause = Utils.damageCauseMap.get(e.getCause());
				int amountd = (int) (e.getFinalDamage() / 2);
				amountd = (int) ((amountd * 100) / 100.0);
				String amounts = String.valueOf(amountd);
				if (!(Objects.equals(e.getCause().toString(), "ENTITY_ATTACK")
						|| Objects.equals(e.getCause().toString(), "ENTITY_EXPLOSION")
						|| Objects.equals(e.getCause().toString(), "ENTITY_SWEEP_ATTACK")
						|| Objects.equals(e.getCause().toString(), "PROJECTILE"))
				) {
					if (amountd > 0.01) {
						Utils.atAll(Utils.dmgMessage(e.getEntity().getName(), amounts, cause));
					}
				}
			}
		}
	}

	@EventHandler
	public void onDmg(EntityDamageByEntityEvent e){
		if (Utils.dmgListenerToggle) {
			String attacker = e.getDamager().getName().replace('_',' ');
			if (e.getEntity() instanceof Player) {
				double damageAmount = e.getFinalDamage() / 2;
				damageAmount = (damageAmount * 100) / 100.0;
				String amounts = String.valueOf(damageAmount);
				if (damageAmount > 0.01) {
					Utils.atAll(Utils.dmgMessage(e.getEntity().getName(), amounts, attacker));
				}
			}
		}
	}
}

File: src/main/java/de/aquafun3d/challenge/listeners/DmgListener.kt
package de.aquafun3d.challenge.listeners

import de.aquafun3d.challenge.Main
import de.aquafun3d.challenge.utils.Utils
import org.bukkit.Bukkit
import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.entity.EntityDamageByEntityEvent
import org.bukkit.event.entity.EntityDamageEvent
import kotlin.math.roundToInt

class DmgListener : Listener {

	//handles damage received by anything else then other entity's
	@EventHandler
	fun onDmg(e: EntityDamageEvent) {
		if (e.entity is Player) {
			e.damage = e.damage * Main.settingsInv?.getDmg()!!
			if (Utils.dmgListenerToggle) {
				val cause: String? = Utils.damageCauseMap[e.cause]
				var amountd = e.finalDamage / 2
				amountd = (amountd * 100).roundToInt() / 100.0
				val amounts = amountd.toString()
				if (!(e.cause.toString() == "ENTITY_ATTACK"
							|| e.cause.toString() == "ENTITY_EXPLOSION"
							|| e.cause.toString() == "ENTITY_SWEEP_ATTACK"
							|| e.cause.toString() == "PROJECTILE")
				) {
					if (amountd > 0.01) {
						Bukkit.getOnlinePlayers().forEach { p: Player? -> p!!.sendMessage(Utils.dmgMessage(e.entity.name, amounts, cause!!)) }
					}
				}
			}
		}
	}

	@EventHandler
	fun onDmg(e: EntityDamageByEntityEvent){
		if (Utils.dmgListenerToggle) {
			val attacker: String = e.damager.name.replace('_',' ')
			if (e.entity is Player) {
				var damageAmount: Double = e.finalDamage / 2
				damageAmount = (damageAmount * 100).roundToInt() / 100.0
				val amounts = damageAmount.toString()
				if (damageAmount > 0.01) {
					Bukkit.getOnlinePlayers().forEach { p: Player? -> p!!.sendMessage(Utils.dmgMessage(e.entity.name, amounts, attacker)) }
				}
			}
		}
	}
}